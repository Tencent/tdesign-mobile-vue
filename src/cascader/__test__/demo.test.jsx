/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/cascader/demos/base.vue';
import mobileVue from '@/cascader/demos/mobile.vue';

const mapper = {
  baseVue,
  mobileVue,
};

describe('Cascader', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Cascader ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
