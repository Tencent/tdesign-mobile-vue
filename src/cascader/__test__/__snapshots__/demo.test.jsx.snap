// Vitest Snapshot v1

exports[`Cascader > Cascader baseVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="t-cell t-cell--middle t-cell--bordered t-input t-input--size-small"
  >
    <div
      class="t-cell__left-icon"
    >
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__title"
    >
      
      <div
        class="t-input--label"
      >
        地址
      </div>
      
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__note"
    >
      
      <div
        class="t-input__wrap"
      >
        <input
          autocomplete="Off"
          class="t-input__control"
          maxlength="-1"
          name=""
          placeholder="选择地址"
          readonly=""
          type="text"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      <!--v-if-->
      
    </div>
    <!--v-if-->
  </div>
  <!--teleport start-->
  <div
    class="t-popup"
  >
    <transition-stub
      appear="false"
      css="true"
      name="fade"
      persisted="false"
    >
      <transition-stub
        appear="false"
        css="true"
        name="t-overlay"
        persisted="true"
      >
        <div
          class="t-overlay"
          style="z-index: 1000; transition-duration: 300ms; display: none;"
        >
          
          
        </div>
      </transition-stub>
    </transition-stub>
    <transition-stub
      appear="false"
      css="true"
      name="slide-bottom"
      persisted="true"
    >
      <div
        class="t-popup--content t-popup--content-bottom"
        style="display: none;"
      >
        
        <div
          class="t-cascader__wrapper"
        >
          <div
            class="t-cascader__title"
          >
            
            选择地址
            
          </div>
          <div
            class="t-cascader__close-btn"
          >
            <svg
              class="t-icon t-icon-close"
              fill="none"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
          <div
            class="t-cascader__content"
          >
            <div
              class="t-cascader__steps"
            >
              
              <div
                class="t-cascader__step"
              >
                <div
                  class="t-cascader__step-dot t-cascader__step-dot--active t-cascader__step-dot--last"
                />
                <div
                  class="t-cascader__step-label t-cascader__step-label--active"
                >
                  选择选项
                </div>
              </div>
              
            </div>
            <div
              class="t-cascader__options-title"
            >
              选项标题
            </div>
            <div
              class="t-cascader__options-content"
            >
              
              <div
                style="width: 100%;"
              >
                <transition-stub
                  appear="true"
                  css="true"
                  name="slide"
                  persisted="true"
                >
                  <div
                    style="width: 100%;"
                  >
                    
                    <div
                      class="t-cascader-item"
                    >
                      <div
                        class="t-cascader-item-content"
                      >
                        <!-- &lt;t-node v-if="!(typeof labelTNode === 'string')" :content="labelTNode"&gt;&lt;/t-node&gt; -->
                        广东省
                      </div>
                      <div
                        class="t-cascader-item-active-icon"
                        style="display: none;"
                      >
                        <svg
                          class="t-icon t-icon-check"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </div>
                    </div>
                    <div
                      class="t-cascader-item"
                    >
                      <div
                        class="t-cascader-item-content"
                      >
                        <!-- &lt;t-node v-if="!(typeof labelTNode === 'string')" :content="labelTNode"&gt;&lt;/t-node&gt; -->
                        福建省
                      </div>
                      <div
                        class="t-cascader-item-active-icon"
                        style="display: none;"
                      >
                        <svg
                          class="t-icon t-icon-check"
                          fill="none"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                        >
                          <path
                            d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                            fill="currentColor"
                            fill-opacity="0.9"
                          />
                        </svg>
                      </div>
                    </div>
                    
                  </div>
                </transition-stub>
              </div>
              
            </div>
          </div>
        </div>
        
      </div>
    </transition-stub>
  </div>
  <!--teleport end-->
  
</div>
`;

exports[`Cascader > Cascader mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Cascader 级联选择器
  </h1>
  <p
    class="summary"
  >
    用于多层级数据选择，主要为树形结构，可展示更多的数据
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      01 基础
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      基本用法
    </p>
    
    
    <div
      class="t-cell t-cell--middle t-cell--bordered t-input t-input--size-small"
    >
      <div
        class="t-cell__left-icon"
      >
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="t-cell__title"
      >
        
        <div
          class="t-input--label"
        >
          地址
        </div>
        
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="t-cell__note"
      >
        
        <div
          class="t-input__wrap"
        >
          <input
            autocomplete="Off"
            class="t-input__control"
            maxlength="-1"
            name=""
            placeholder="选择地址"
            readonly=""
            type="text"
          />
          <!--v-if-->
          <!--v-if-->
        </div>
        <!--v-if-->
        
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <div
      class="t-popup"
    >
      <transition-stub
        appear="false"
        css="true"
        name="fade"
        persisted="false"
      >
        <transition-stub
          appear="false"
          css="true"
          name="t-overlay"
          persisted="true"
        >
          <div
            class="t-overlay"
            style="z-index: 1000; transition-duration: 300ms; display: none;"
          >
            
            
          </div>
        </transition-stub>
      </transition-stub>
      <transition-stub
        appear="false"
        css="true"
        name="slide-bottom"
        persisted="true"
      >
        <div
          class="t-popup--content t-popup--content-bottom"
          style="display: none;"
        >
          
          <div
            class="t-cascader__wrapper"
          >
            <div
              class="t-cascader__title"
            >
              
              选择地址
              
            </div>
            <div
              class="t-cascader__close-btn"
            >
              <svg
                class="t-icon t-icon-close"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <div
              class="t-cascader__content"
            >
              <div
                class="t-cascader__steps"
              >
                
                <div
                  class="t-cascader__step"
                >
                  <div
                    class="t-cascader__step-dot t-cascader__step-dot--active t-cascader__step-dot--last"
                  />
                  <div
                    class="t-cascader__step-label t-cascader__step-label--active"
                  >
                    选择选项
                  </div>
                </div>
                
              </div>
              <div
                class="t-cascader__options-title"
              >
                选项标题
              </div>
              <div
                class="t-cascader__options-content"
              >
                
                <div
                  style="width: 100%;"
                >
                  <transition-stub
                    appear="true"
                    css="true"
                    name="slide"
                    persisted="true"
                  >
                    <div
                      style="width: 100%;"
                    >
                      
                      <div
                        class="t-cascader-item"
                      >
                        <div
                          class="t-cascader-item-content"
                        >
                          <!-- &lt;t-node v-if="!(typeof labelTNode === 'string')" :content="labelTNode"&gt;&lt;/t-node&gt; -->
                          广东省
                        </div>
                        <div
                          class="t-cascader-item-active-icon"
                          style="display: none;"
                        >
                          <svg
                            class="t-icon t-icon-check"
                            fill="none"
                            height="1em"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                              fill="currentColor"
                              fill-opacity="0.9"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="t-cascader-item"
                      >
                        <div
                          class="t-cascader-item-content"
                        >
                          <!-- &lt;t-node v-if="!(typeof labelTNode === 'string')" :content="labelTNode"&gt;&lt;/t-node&gt; -->
                          福建省
                        </div>
                        <div
                          class="t-cascader-item-active-icon"
                          style="display: none;"
                        >
                          <svg
                            class="t-icon t-icon-check"
                            fill="none"
                            height="1em"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                              fill="currentColor"
                              fill-opacity="0.9"
                            />
                          </svg>
                        </div>
                      </div>
                      
                    </div>
                  </transition-stub>
                </div>
                
              </div>
            </div>
          </div>
          
        </div>
      </transition-stub>
    </div>
    <!--teleport end-->
    
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      02 状态
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      禁用可选项
    </p>
    
    <div
      class="t-cell t-cell--middle t-cell--bordered t-input t-input--size-small"
    >
      <div
        class="t-cell__left-icon"
      >
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="t-cell__title"
      >
        
        <div
          class="t-input--label"
        >
          地址
        </div>
        
        <!--v-if-->
        <!--v-if-->
      </div>
      <div
        class="t-cell__note"
      >
        
        <div
          class="t-input__wrap"
        >
          <input
            autocomplete="Off"
            class="t-input__control"
            maxlength="-1"
            name=""
            placeholder="选择地址"
            readonly=""
            type="text"
          />
          <!--v-if-->
          <!--v-if-->
        </div>
        <!--v-if-->
        
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <div
      class="t-popup"
    >
      <transition-stub
        appear="false"
        css="true"
        name="fade"
        persisted="false"
      >
        <transition-stub
          appear="false"
          css="true"
          name="t-overlay"
          persisted="true"
        >
          <div
            class="t-overlay"
            style="z-index: 1000; transition-duration: 300ms; display: none;"
          >
            
            
          </div>
        </transition-stub>
      </transition-stub>
      <transition-stub
        appear="false"
        css="true"
        name="slide-bottom"
        persisted="true"
      >
        <div
          class="t-popup--content t-popup--content-bottom"
          style="display: none;"
        >
          
          <div
            class="t-cascader__wrapper"
          >
            <div
              class="t-cascader__title"
            >
              
              标题
              
            </div>
            <div
              class="t-cascader__close-btn"
            >
              <svg
                class="t-icon t-icon-close"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <div
              class="t-cascader__content"
            >
              <div
                class="t-cascader__steps"
              >
                
                <div
                  class="t-cascader__step"
                >
                  <div
                    class="t-cascader__step-dot t-cascader__step-dot--active t-cascader__step-dot--last"
                  />
                  <div
                    class="t-cascader__step-label t-cascader__step-label--active"
                  >
                    选择选项
                  </div>
                </div>
                
              </div>
              <div
                class="t-cascader__options-title"
              >
                选项标题
              </div>
              <div
                class="t-cascader__options-content"
              >
                
                <div
                  style="width: 100%;"
                >
                  <transition-stub
                    appear="true"
                    css="true"
                    name="slide"
                    persisted="true"
                  >
                    <div
                      style="width: 100%;"
                    >
                      
                      <div
                        class="t-cascader-item disabled"
                      >
                        <div
                          class="t-cascader-item-content"
                        >
                          <!-- &lt;t-node v-if="!(typeof labelTNode === 'string')" :content="labelTNode"&gt;&lt;/t-node&gt; -->
                          浙江省
                        </div>
                        <div
                          class="t-cascader-item-active-icon"
                          style="display: none;"
                        >
                          <svg
                            class="t-icon t-icon-check"
                            fill="none"
                            height="1em"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                              fill="currentColor"
                              fill-opacity="0.9"
                            />
                          </svg>
                        </div>
                      </div>
                      <div
                        class="t-cascader-item"
                      >
                        <div
                          class="t-cascader-item-content"
                        >
                          <!-- &lt;t-node v-if="!(typeof labelTNode === 'string')" :content="labelTNode"&gt;&lt;/t-node&gt; -->
                          江苏省
                        </div>
                        <div
                          class="t-cascader-item-active-icon"
                          style="display: none;"
                        >
                          <svg
                            class="t-icon t-icon-check"
                            fill="none"
                            height="1em"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M6.43 9.92l6.23-6.22.92.92-7.15 7.14L1.97 7.3l.92-.92 3.54 3.54z"
                              fill="currentColor"
                              fill-opacity="0.9"
                            />
                          </svg>
                        </div>
                      </div>
                      
                    </div>
                  </transition-stub>
                </div>
                
              </div>
            </div>
          </div>
          
        </div>
      </transition-stub>
    </div>
    <!--teleport end-->
    
  </div>
</div>
`;
